File: cfnsane/resources/bucket.py
from troposphere.s3 import Bucket as TSBucket

from cfnsane import NAMESPACE, Resource

class Bucket(TSBucket, Resource):

    resource_type: str = f"{NAMESPACE}::S3::Bucket"

    def __init__(self, *args, **kwargs):
        kwargs["PublicAccessBlockConfiguration"] = {
            "BlockPublicAcls": True,
            "BlockPublicPolicy": True,
            "IgnorePublicAcls": True,
            "RestrictPublicBuckets": True,
        }

        super().__init__(*args, **kwargs)

File: lib/s3/bucket.js
const s3 = require('@aws-cdk/aws-s3');
const Aspect = require('../aspect');

class Bucket extends Aspect{
    resource = s3.CfnBucket;

    blockPublicAccess(node) {
        node.publicAccessBlockConfiguration = new s3.BlockPublicAccess({
            blockPublicAcls: true,
            blockPublicPolicy: true,
            ignorePublicAcls: true,
            restrictPublicBuckets: true,
        });
    }
    sanity(node) {
        if (!node.publicAccessBlockConfiguration){
            this.blockPublicAccess(node)
        }
    }
}

module.exports = Bucket;
